### ISSUE TITLE
Load Remote Deck from URL (in Deck Editor)

### ISSUE THREAD

**Issue by Omicron7** [Cockatrice/Cockatrice#4032]
**Description of New Feature**
I would like the ability to Load a Remote Deck from a URL instead of just my filesystem, or the remote server. It should be implemented to allow any format Cockatrice already supports, defaulting to XML.

**Additional Context**
This could allow loading decks from a shared library (github/dropbox/drive/...) or from an API that returns a deck in XML, text or another supported format (based on Content-Type headers).

I'm looking forward to the upcoming [JumpStart](https://mtg.gamepedia.com/Jumpstart) and was thinking of creating a small API that would return a Deck in XML that is a random set of 2 or more of the 121 possible packs. Being able to "Load Remote Deck from URL" would be a an easy and quick way to play JumpStart in Cockatrice.

**Similar Requests**
I searched around and didn't seen any existing issues related or similar to this.

**Comment by ZeldaZach** [comment]
I like the idea, but this will add a lot of work on our side to ensure we stay compatible with upstream providers. We already have integration with a few providers, so I could see about supporting TappedOut and DeckStats for remote loading. I'm sure people will want to see Scryfall integration as well.

**Comment by Omicron7** [comment]
I was thinking of this feature primarily from a game Lobby perspective. You're in a game, and you can load a random deck you've never seen and click "Ready". I'm assuming the integration with those providers is more on the Create a deck/deck editor side of things.

**Comment by ebbit1q** [comment]
How would this be more convenient for users than say a load deck from clipboard inside the lobby?

**Comment by Omicron7** [comment]
@ebbit1q I agree. The more I've thought through this and tested out my PR with existing deck management/draft sites, the more I'm leaning toward just a basic load from clipboard. It'll be much less confusing to end users than figuring out the URL to use.

**Comment by ebbit1q** [comment]
A way I do see this working is if you were to save the url inside a decklist file and have it fetch the list when you load it, that way you could change the list on that website and as long as the url stays the same it'd update.

**Comment by ZeldaZach** [comment]
After talking with @ebbit1q we've decided this is probably not the way we want to go. Having a way to load a deck from a URL in the deck editor is more in line with what we were thinking, compared to more buttons in the game menu where decks could have some weird formatting issues.

**Comment by ebbit1q** [comment]
Maybe instead add the upstream url to a .cod file which'll fetch the decklist on demand?
